src/AzureRenderManager/WebApp/Views/Reporting/Index.cshtml

```diff
@@ -50,6 +53,103 @@
             continue;
         }
 
+        var id = Guid.NewGuid().ToString();
+        var cols = data.Properties.Columns;
+        var dateIndex = cols.FindIndex(col => col.Name == "UsageDate");
+        var costIndex = cols.FindIndex(col => col.Name == "PreTaxCost");
+        var currencyIndex = cols.FindIndex(col => col.Name == "Currency");
+        var meterCategoryIndex = cols.FindIndex(col => col.Name == "MeterSubCategory");
+        var currency = (string)data.Properties.Rows.First()[currencyIndex]; // assumption: all values have same currency
+        
+        <canvas id="graph-@id" ></canvas>
+
+        <script type="text/javascript">
+            var ctx = document.getElementById("graph-@id").getContext('2d');
+            ctx.canvas.width = 800;
+            ctx.canvas.height = 400;
+
+            var datasets = [
+            @foreach (var meterCategory in data.Properties.Rows.GroupBy(row => (string)row[meterCategoryIndex]))
```

Comment:
I wonder if this logic should be moved server side?  It's quite difficult to follow with the JS/HTML in razor view.

Reply:
I've separated it out more so the templating/generation of values is not interleaved - this should be much clearer.